packages/edit-widgets/src/store/actions.js

```diff
@@ -168,15 +184,32 @@ export function* saveWidgetArea( widgetAreaId ) {
 			clientId: block.clientId,
 		} );
 	}
+	for ( let i = 0; i < deletedWidgets.length; i++ ) {
```

Comment:
Since we're not using `i` outside of `deletedWidgets[ i ]`, this can be a `for ... of` loop.